narwhals/__init__.py

```diff
@@ -81,7 +81,7 @@
 from narwhals.utils import maybe_reset_index
 from narwhals.utils import maybe_set_index
 
-__version__ = "1.38.0"
+__version__ = "1.38.1"
```

Comment:
Completely unrelated from this specific release but, should we use:

```py
from importlib import metadata

__version__ = metadata.version(__name__)
```
to simplify the sync with pyproject.toml?

Reply:
i remember there being some issue with importlib on older python versions, but if we're past those and don't require backports then sure 👍